File: programs/vertex-program/src/states/indexer.rs
use anchor_lang::prelude::*;

#[account]
#[derive(Debug, InitSpace)]
pub struct Indexer {
  pub owner: Pubkey,
  pub bump: u8,
  pub indexer_id: u64,
  pub price_per_gb_lamports: u64,
  pub rent_lamports: u64,
}

impl Indexer {
  pub fn init(
    &mut self,
    indexer_id: u64,
    price_per_gb_lamports: u64,
    owner: Pubkey,
    bump: u8,
    rent_lamports: u64,
  ) {
    self.indexer_id = indexer_id;
    self.price_per_gb_lamports = price_per_gb_lamports;
    self.owner = owner;
    self.bump = bump;
    self.rent_lamports = rent_lamports;
  }
}

File: sdk/state/indexer.ts
import { PublicKey } from "@solana/web3.js";
import BN from "bn.js";
import { Display } from "./display";

export interface IIndexer {
  owner: PublicKey;
  bump: number;
  indexerId: BN;
  pricePerGbLamports: BN;
  rentLamports: BN;
}

export class Indexer implements Display {
  state: IIndexer;
  
  constructor(data: IIndexer) {
    this.state = data;
  }

  display(): void {
    console.log(
      [
        "Indexer {",
        `  owner              : ${this.state.owner.toString()}`,
        `  bump               : ${this.state.bump}`,
        `  indexerId          : ${this.state.indexerId.toString()}`,
        `  pricePerGbLamports : ${this.state.pricePerGbLamports.toString()}`,
        `  rentLamports       : ${this.state.rentLamports.toString()}`,
        "}",
      ].join("\n")
    );
  }
}
